### ISSUE TITLE
Refer or integrate mobilenet-v3-rs into examples

### ISSUE THREAD

**Issue by jerry73204** [LaurentMazare/tch-rs#67]
It happens that I had to implement MobileNetV3 ([repo link](https://github.com/jerry73204/mobilenet-v3-rs)) for some purpose. I think it's small enough to fit into examples. It provides an example program and library interface. I have plan A and B to publish this work.

* A: Integrate into the examples. It would be better to let users import the model in the code.
* B: Keep it a standalone repo, and I will publish it to crates.io.

If plan A, we would need someone else to verify the correctness of the impl. So far I'm only certain that the precision reaches paper's claim in minutes.

**Comment by LaurentMazare** [comment]
This seems very nice. We only have mobilenet v2 currently in the examples directory ([mobilenet.rs](https://github.com/LaurentMazare/tch-rs/blob/master/src/vision/mobilenet.rs)). Having mobilenet v3 would sound like a nice addition if you want to go the A way.
Do you know if there are some pretrained weights for the original model ? This could make it easier to verify correctness + this way it could directly be supported for things like the [inference example](https://github.com/LaurentMazare/tch-rs/blob/master/examples/pretrained-models/main.rs).

**Comment by jerry73204** [comment]
The paper is shinny new and only handful of MNetV3 impls out there, and yes, I agree pretrained weight is good source for verification. That implies some work, so I haven't put it into my schedule. I would compete the performance b/w existing MNetV2 and V3 to see if it really works.

For MNIST, the V3 reaches 98% accuracy on train and test set. For CIFAR-10, the accuracy sways near by 96%, and gets 75% on test set. It's clear clue of overfitting. I looked into existing V2. I gets 96% most of time, and 68% on on test set. It overfits as well, and V3 did some improvement.

We could give plan A a chance. I prefer some customizability for users, so there are things to talk about:
* Let users specify model parameters like input channels, dropout, width multi and n. of classes. Unlike existing V2 we can only specify n. of classes.
* Make building blocks available to users. We have bottleneck module, squeeze-excitation module, and {,H-}swish activation. If you think it's too verbose,  provide {,H-}swish because it's already in PyTorch.

**Comment by LaurentMazare** [comment]
Thanks, happy to help with importing some pretrained weights if there is already a good PyTorch implementation somewhere.
Re cifar10 I'm probably missing something but 75% test accuracy seems pretty low even with some strong overfitting. Looking at existing implementations for v2, it should be more than 94% accuracy on test/validation, e.g. https://github.com/tinyalpha/mobileNet-v2_cifar10 or https://github.com/kuangliu/pytorch-cifar
Also the models in examples/cifar also reach 94% test accuracy (and 99% on the training set). Is there something that I'm missing here ?

**Comment by jerry73204** [comment]
That's weird. Even changing the example to train existing v2 model does not reach that high accuracy. In the recent test I got 73% on val set, with batch size 32 and 100 epochs. Of course, v3 has no luck. The only catch is that there is no shuffling. Not sure if it crucial part and I'll give a try.

**Comment by LaurentMazare** [comment]
Closing as this has stalled a bit, feel free to re-open if you want to push more on this (it would certainly make a nice additional example).
